functionary/ui/views/teams.py

```diff
@@ -1,14 +1,26 @@
+from django.contrib.auth.mixins import UserPassesTestMixin
 from django.views.generic.detail import DetailView
 from django.views.generic.list import ListView
 
-from core.models import Team
+from core.auth import Permission
+from core.models import Team, TeamUserRole
 
 
 class TeamListView(ListView):
     model = Team
 
+    def get_queryset(self):
+        """Sorts based on team name, then env name."""
+        if self.request.user.is_superuser:
+            return super().get_queryset().order_by("name")
+        else:
+            teams = TeamUserRole.objects.filter(user=self.request.user).values(
```

Comment:
I guess this is similar to the `EnvironmentListView` situation.  But also I at some point suggested that the way you've got the environments and teams displaying in the UI currently isn't really what we we're going to want, so before you change anything we should probably talk about if those things just go away for now.